[32mINFO    [0m juju.model:model.py:2971 Waiting for model:
  shard-one/0 [idle] active: Primary
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  shard-two/1 [idle] active: 
  shard-two/2 [idle] active: 
  config-server/0 [idle] active: Primary
  config-server/1 [idle] active: 
  config-server/2 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:903 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.6.1    unsupported  02:27:29Z

App            Version  Status  Scale  Charm    Channel  Rev  Exposed  Message
config-server           active      3  mongodb             0  no       
shard-one               active      3  mongodb             1  no       
shard-two               active      3  mongodb             2  no       

Unit              Workload  Agent  Machine  Public address  Ports            Message
config-server/0   active    idle   0        10.164.85.186   27017-27018/tcp  Primary
config-server/1*  active    idle   1        10.164.85.170   27017-27018/tcp  
config-server/2   active    idle   2        10.164.85.5     27017-27018/tcp  
shard-one/0       active    idle   3        10.164.85.253   27017/tcp        Primary
shard-one/1*      active    idle   4        10.164.85.172   27017/tcp        
shard-one/2       active    idle   5        10.164.85.118   27017/tcp        
shard-two/0       active    idle   6        10.164.85.10    27017/tcp        Primary
shard-two/1*      active    idle   7        10.164.85.93    27017/tcp        
shard-two/2       active    idle   8        10.164.85.20    27017/tcp        

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.164.85.186  juju-f6561b-0  ubuntu@22.04      Running
1        started  10.164.85.170  juju-f6561b-1  ubuntu@22.04      Running
2        started  10.164.85.5    juju-f6561b-2  ubuntu@22.04      Running
3        started  10.164.85.253  juju-f6561b-3  ubuntu@22.04      Running
4        started  10.164.85.172  juju-f6561b-4  ubuntu@22.04      Running
5        started  10.164.85.118  juju-f6561b-5  ubuntu@22.04      Running
6        started  10.164.85.10   juju-f6561b-6  ubuntu@22.04      Running
7        started  10.164.85.93   juju-f6561b-7  ubuntu@22.04      Running
8        started  10.164.85.20   juju-f6561b-8  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:909 Juju error logs:

machine-7: 01:51:37 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-7: 01:51:37 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-two-1: 01:51:37 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-6: 01:51:37 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-6: 01:51:37 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-two-0: 01:51:38 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:51:49 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:51:49 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-one-1: 01:51:49 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-8: 01:51:57 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-8: 01:51:57 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-two-2: 01:51:57 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-shard-two-1: 01:52:00 ERROR unit.shard-two/1.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-two-1: 01:52:00 ERROR unit.shard-two/1.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-shard-two-0: 01:52:01 ERROR unit.shard-two/0.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-two-0: 01:52:01 ERROR unit.shard-two/0.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
machine-1: 01:52:06 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:52:06 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-config-server-1: 01:52:06 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-shard-one-1: 01:52:12 ERROR unit.shard-one/1.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-one-1: 01:52:12 ERROR unit.shard-one/1.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-shard-two-2: 01:52:19 ERROR unit.shard-two/2.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-two-2: 01:52:19 ERROR unit.shard-two/2.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
machine-5: 01:52:22 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:52:22 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-one-2: 01:52:22 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:52:25 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:52:25 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-config-server-0: 01:52:25 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:52:26 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:52:26 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-shard-one-0: 01:52:26 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-config-server-1: 01:52:27 ERROR unit.config-server/1.juju-log Unable to set params: ['vm.max_map_count']
unit-config-server-1: 01:52:27 ERROR unit.config-server/1.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-shard-one-2: 01:52:45 ERROR unit.shard-one/2.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-one-2: 01:52:45 ERROR unit.shard-one/2.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-config-server-0: 01:52:48 ERROR unit.config-server/0.juju-log Unable to set params: ['vm.max_map_count']
unit-config-server-0: 01:52:48 ERROR unit.config-server/0.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-shard-one-0: 01:52:48 ERROR unit.shard-one/0.juju-log Unable to set params: ['vm.max_map_count']
unit-shard-one-0: 01:52:48 ERROR unit.shard-one/0.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
machine-2: 01:52:53 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:52:53 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-config-server-2: 01:52:53 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-config-server-2: 01:53:09 ERROR unit.config-server/2.juju-log Unable to set params: ['vm.max_map_count']
unit-config-server-2: 01:53:09 ERROR unit.config-server/2.juju-log Error setting values on sysctl: Unable to set params: ['vm.max_map_count']
unit-shard-one-1: 01:53:58 ERROR unit.shard-one/1.juju-log _on_secret_remove: Secret secret:cu84v8kgtj8cu6mpltug seems to have no observers, could be removed
unit-config-server-1: 01:54:12 ERROR unit.config-server/1.juju-log _on_secret_remove: Secret secret:cu84vc4gtj8cu6mpltv0 seems to have no observers, could be removed
unit-shard-two-1: 01:55:42 ERROR unit.shard-two/1.juju-log _on_secret_remove: Secret secret:cu84vh4gtj8cu6mpltvg seems to have no observers, could be removed
unit-config-server-1: 02:00:09 ERROR unit.config-server/1.juju-log config-server:6: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1737511209, 3), 'signature': {'hash': b'\\xe5\\x94g<\\x8d\\xbcA\\xe1\\xc7\\x88\\xc6\\xfc\\xee\\x94vp\\xf2\\x19\\xbf\\xff', 'keyId': 7462552229950521362}}, 'operationTime': Timestamp(1737511209, 3)}")
unit-config-server-1: 02:00:10 ERROR unit.config-server/1.juju-log config-server:6: shard-one shard does not have the same auth as the config server.
unit-config-server-1: 02:00:27 ERROR unit.config-server/1.juju-log config-server:6: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1737511226, 1), \'signature\': {\'hash\': b\'BzW\\xb4\\xcc\\x08\\x82\\x07\\xfc\\x8f\\xcb\\xa4\\xf8\\x1d\\x8e\\xe6\\x00Y\\x08\\r\', \'keyId\': 7462552229950521362}}, \'operationTime\': Timestamp(1737511226, 1)}')
unit-config-server-1: 02:00:27 ERROR unit.config-server/1.juju-log config-server:6: Failed to add shard-one to cluster
unit-config-server-1: 02:00:27 ERROR unit.config-server/1.juju-log config-server:6: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1737511226, 1), \'signature\': {\'hash\': b\'BzW\\xb4\\xcc\\x08\\x82\\x07\\xfc\\x8f\\xcb\\xa4\\xf8\\x1d\\x8e\\xe6\\x00Y\\x08\\r\', \'keyId\': 7462552229950521362}}, \'operationTime\': Timestamp(1737511226, 1)}')
unit-config-server-1: 02:00:47 ERROR unit.config-server/1.juju-log config-server:7: Failed to add shard shard-two to the config server, error=OperationFailure('Authentication failed., full error: {\'ok\': 0.0, \'errmsg\': \'Authentication failed.\', \'code\': 18, \'codeName\': \'AuthenticationFailed\', \'$clusterTime\': {\'clusterTime\': Timestamp(1737511247, 3), \'signature\': {\'hash\': b\'v\\\'\\x19,\\xf3\\xe2\\x1d"\\x85(\\xaf\\x97p#3\\xac$soX\', \'keyId\': 7462552229950521362}}, \'operationTime\': Timestamp(1737511247, 3)}')
unit-config-server-1: 02:00:48 ERROR unit.config-server/1.juju-log config-server:7: shard-two shard does not have the same auth as the config server.
unit-config-server-1: 02:01:04 ERROR unit.config-server/1.juju-log config-server:7: Failed to add shard shard-two to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-two, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-two\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1737511264, 2), \'signature\': {\'hash\': b\'J\\xbb\\xad\\x11\\xa8\\xb7\\xa0D\\xfdc\\xb5K\\x19\\xed\\xf4S\\xcf{\\x8e\\x8f\', \'keyId\': 7462552229950521362}}, \'operationTime\': Timestamp(1737511264, 2)}')
unit-config-server-1: 02:01:04 ERROR unit.config-server/1.juju-log config-server:7: Failed to add shard-two to cluster
unit-config-server-1: 02:01:04 ERROR unit.config-server/1.juju-log config-server:7: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-two, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-two\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1737511264, 2), \'signature\': {\'hash\': b\'J\\xbb\\xad\\x11\\xa8\\xb7\\xa0D\\xfdc\\xb5K\\x19\\xed\\xf4S\\xcf{\\x8e\\x8f\', \'keyId\': 7462552229950521362}}, \'operationTime\': Timestamp(1737511264, 2)}')
unit-shard-two-1: 02:07:23 ERROR unit.shard-two/1.juju-log _on_secret_remove: Secret secret:cu84vh4gtj8cu6mpltvg seems to have no observers, could be removed
unit-config-server-1: 02:26:07 ERROR unit.config-server/1.juju-log Cannot proceed with refresh. Failed to check cluster health, error: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 6790573ef3ff8b96d5c115c0, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('10.164.85.118', 27017) server_type: RSSecondary, rtt: 0.0008227829998759262>, <ServerDescription ('10.164.85.172', 27017) server_type: RSSecondary, rtt: 0.0014126629998827411>, <ServerDescription ('10.164.85.253', 27017) server_type: Unknown, rtt: None>]>
unit-config-server-1: 02:26:10 ERROR unit.config-server/1.juju-log Cannot proceed with refresh. Failed to check cluster health, error: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 67905741f3ff8b96d5c115c5, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('10.164.85.118', 27017) server_type: RSSecondary, rtt: 0.0008937359998526517>, <ServerDescription ('10.164.85.172', 27017) server_type: RSSecondary, rtt: 0.0006890250001561071>, <ServerDescription ('10.164.85.253', 27017) server_type: Unknown, rtt: None>]>
unit-config-server-1: 02:26:13 ERROR unit.config-server/1.juju-log Cannot proceed with refresh. Failed to check cluster health, error: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 67905744f3ff8b96d5c115ca, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('10.164.85.118', 27017) server_type: RSSecondary, rtt: 0.0006590089997189352>, <ServerDescription ('10.164.85.172', 27017) server_type: RSSecondary, rtt: 0.0009509129999969446>, <ServerDescription ('10.164.85.253', 27017) server_type: Unknown, rtt: None>]>
unit-config-server-1: 02:26:42 ERROR unit.config-server/1.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-1/charm/src/charm.py", line 1618, in <module>
    main(MongodbOperatorCharm)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/main.py", line 551, in main
    manager.run()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/main.py", line 530, in run
    self._emit()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/main.py", line 519, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/main.py", line 147, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 348, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 860, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 950, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-config-server-1/charm/src/upgrades/mongodb_upgrade.py", line 148, in _on_pre_upgrade_check_action
    self._upgrade.pre_upgrade_check()
  File "/var/lib/juju/agents/unit-config-server-1/charm/lib/charms/mongodb/v0/upgrade_helpers.py", line 336, in pre_upgrade_check
    if not self._charm.upgrade.is_cluster_able_to_read_write():
  File "/var/lib/juju/agents/unit-config-server-1/charm/lib/charms/mongodb/v0/upgrade_helpers.py", line 523, in is_cluster_able_to_read_write
    return self.is_sharded_cluster_able_to_read_write()
  File "/var/lib/juju/agents/unit-config-server-1/charm/lib/charms/mongodb/v0/upgrade_helpers.py", line 539, in is_sharded_cluster_able_to_read_write
    write_replicated = self.is_write_on_secondaries(
  File "/var/lib/juju/agents/unit-config-server-1/charm/lib/charms/mongodb/v0/upgrade_helpers.py", line 651, in is_write_on_secondaries
    self.confirm_excepted_write_on_replica(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 475, in __call__
    do = self.iter(retry_state=retry_state)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 418, in exc_check
    raise retry_exc.reraise()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 185, in reraise
    raise self.last_attempt.result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/tenacity/__init__.py", line 478, in __call__
    result = fn(*args, **kwargs)
  File "/var/lib/juju/agents/unit-config-server-1/charm/lib/charms/mongodb/v0/upgrade_helpers.py", line 610, in confirm_excepted_write_on_replica
    if query[0][WRITE_KEY] != expected_write_value:
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/cursor.py", line 638, in __getitem__
    for doc in clone:  # type: ignore[attr-defined]
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/cursor.py", line 1281, in __next__
    return self.next()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/cursor.py", line 1257, in next
    if len(self._data) or self._refresh():
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/cursor.py", line 1205, in _refresh
    self._send_message(q)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/cursor.py", line 1100, in _send_message
    response = client._run_operation(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/_csot.py", line 119, in csot_wrapper
    return func(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 1754, in _run_operation
    return self._retryable_read(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 1863, in _retryable_read
    return self._retry_internal(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/_csot.py", line 119, in csot_wrapper
    return func(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 1830, in _retry_internal
    ).run()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 2554, in run
    return self._read() if self._is_read else self._write()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 2689, in _read
    self._server = self._get_server()
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 2645, in _get_server
    return self._client._select_server(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/mongo_client.py", line 1649, in _select_server
    server = topology.select_server(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/topology.py", line 398, in select_server
    server = self._select_server(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/topology.py", line 376, in _select_server
    servers = self.select_servers(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/topology.py", line 283, in select_servers
    server_descriptions = self._select_servers_loop(
  File "/var/lib/juju/agents/unit-config-server-1/charm/venv/lib/python3.10/site-packages/pymongo/synchronous/topology.py", line 333, in _select_servers_loop
    raise ServerSelectionTimeoutError(
pymongo.errors.ServerSelectionTimeoutError: No servers found yet, Timeout: 1.0s, Topology Description: <TopologyDescription id: 67905760f3ff8b96d5c115e0, topology_type: Single, servers: [<ServerDescription ('10.164.85.253', 27017) server_type: Unknown, rtt: None>]>
machine-3: 02:26:56 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: api connection broken unexpectedly
machine-3: 02:26:56 ERROR juju.api.watcher error trying to stop watcher: websocket: close sent

[32mINFO    [0m pytest_operator.plugin:plugin.py:991 Forgetting model main...